### ISSUE TITLE
Implementing 'findByUuidOrFail' method

### ISSUE THREAD

**Issue by majidalaeinia** [binarycabin/Laravel-UUID#5]
As you might have read on [Laravel Documentation](https://laravel.com/docs/5.8/eloquent#retrieving-single-models):

> Sometimes you may wish to throw an exception if a model is not found. This is particularly useful  in routes or controllers. The `findOrFail` and `firstOrFail` methods will retrieve the first result of the query; however, if no result is found, a `Illuminate\Database\Eloquent\ModelNotFoundException` will be thrown:
> 
> ```php
> $model = App\Flight::findOrFail(1);
> 
> $model = App\Flight::where('legs', '>', 100)->firstOrFail();$model = App\Flight::findOrFail(1);
> 
> $model = App\Flight::where('legs', '>', 100)->firstOrFail();
> ```
> If the exception is not caught, a `404` HTTP response is automatically sent back to the user. It is not necessary to write explicit checks to return `404` responses when using these methods:
> ```php
> Route::get('/api/flights/{id}', function ($id) {
>     return App\Flight::findOrFail($id);
> });
> ```

It would be nice if we had the equivalent `findByUuidOrFail` method.

**Comment by kilrizzy** [comment]
Agreed this would be a really nice feature. I can add in when I have time but if anyone wants to PR I'm open to using
